src/Pos/Update/Poll/RollTrans.hs

```diff
@@ -72,13 +73,22 @@ instance MonadPoll m => MonadPoll (RollT m) where
     delConfirmedProposal = lift . delConfirmedProposal
 
     addActiveProposal ps = RollT $ do
+        whenNothingM_ (use unPrevProposersL) $ do
+            prev <- getEpochProposers
+            unPrevProposersL .= Just prev
         insertIfNotExist (hash $ psProposal $ ps) unChangedPropsL getProposal
         addActiveProposal ps
 
     deactivateProposal id = RollT $ do
         insertIfNotExist id unChangedPropsL getProposal
```

Comment:
Ok, I see. Actually it won't harm to save previous proposers anyway (even if proposal from old epoch is deactivated), but it's useless. Please add a comment that we don't need to save proposers, because `deactiveProposal` doesn't modify proposers.